internal/controllers/job_event_reconciler.go

```diff
@@ -0,0 +1,223 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	ocpbuildbuild "github.com/rh-ecosystem-edge/kernel-module-management/internal/build/ocpbuild"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/constants"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/meta"
+	ocpbuildsign "github.com/rh-ecosystem-edge/kernel-module-management/internal/sign/ocpbuild"
+	"golang.org/x/exp/maps"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	JobEventReconcilerName = "JobEvent"
+
+	createdAnnotationKey = "kmm.node.kubernetes.io/created-event-sent"
+)
+
+type jobEvent struct {
+	jobType string
+}
+
+func (b *jobEvent) ReasonCancelled() string {
+	return b.jobType + "Cancelled"
+}
+
+func (b *jobEvent) ReasonCreated() string {
+	return b.jobType + "Created"
+}
+
+func (b *jobEvent) ReasonFailed() string {
+	return b.jobType + "Failed"
+}
+
+func (b *jobEvent) ReasonSucceeded() string {
+	return b.jobType + "Succeeded"
+}
+
+func (b *jobEvent) String() string {
+	return b.jobType
+}
+
+var titler = cases.Title(language.English)
+
+func newJobEvent(jobType string) (*jobEvent, error) {
+	if jobType == "" {
+		return nil, errors.New("jobType cannot be empty")
+	}
+
+	je := &jobEvent{
+		jobType: titler.String(jobType),
+	}
+
+	return je, nil
+}
+
+type JobEventReconciler struct {
```

Comment:
WDYT about renaming this file to build_event_reconciler? And renaming all the internal variables/functions accordingly? I think it is more appropriate for downstream

Reply:
I was wondering about this, and I think it may make cherry-picks slightly more difficult. I called the type `JobEventReconciler` so that it works for both upstream and downstream.

Reply:
I was thinking about renaming it to build_sign_events_reconciler (which will be compatible to both upstream and downstream.  You will still need to manually cherry-pick , since the watch is different anyway

Reply:
This PR now includes https://github.com/kubernetes-sigs/kernel-module-management/pull/589 as well. PTAL.